articles/express/index.md

```diff
@@ -0,0 +1,360 @@
+
+If the file is not in the one directory, then the request is passed onto the next `static()` call. When your app gets a request like `/hello.html`, it returns the hello.html file in the public directory.
+
+### Routing
+Express makes request handling easier by mapping requests to different request handlers. A request handler is a function which handles all the requests to a specific path with a specific HTTP method. 
+
+In the basic example above, we saw how to handle a `GET` request. As an application grows in size the routes grow as well as do the request handlers.
+
+Lets see how we can use [Routers](http://expressjs.com/en/4x/api.html#router) to split a large app into smaller, maintainable functions. 
+
+According to the documentation, a Router is "an isolated instance of middleware and routes. Routers can be thought of as “mini” applications only capable of performing middleware and routing".
+
+Routers can be used like middleware functions, they can be "`.use()`d". A simple example:
+
+```JavaScript
+//app.js the main file
+
+var express = require("express");
+var apiRouter = require("./routes/api_router");
+
+var app = express();
+
+app.use("/api", apiRouter);
+
+app.listen(3000);
+```
+
+```JavaScript
+//routes/api_router.js
+var express = require("express");
+var router = express.Router();
+
+router.get("/route1", function(req, res, next){
+        res.send("Success !!");
+        next();
+});
+
+module.exports = router;
+
+```
+When you start the app and visit the URL `localhost:3000/api/route1` you can see `Success!!`. Take a look at all the router functions [here](http://expressjs.com/en/4x/api.html#router).
+
+#### Useful Routing Tips
+
+**Grabbing route parameters**
+
+Suppose you are building a website for a company that showcases their products, each product has a *productID*. You want the URL for product 1 to be `/product/1`. 
+
+Instead of defining a route for every product, you can define a single route for everything in the form of `product/productID` and then return a file based on the productID.
+
+```JavaScript
+var express = require("express");
+var app = express();
+
+//use a colon to grab a parameter
+
+app.get("/product/:productId", function(req, res){
+    var pid = parseInt(req.params.userid, 10);
+    //res.send   manipulate string to get file with name as productID or something and use a static file server
+    };
+```
+
+**Using Regular Expressions to match routes**
+
+[Regular Expressions (RE)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions) are patterns used to match character combinations in strings. We can use RE to match parameters and define our routes. 
+
+For example, using the example above if we wanted the productId to be only an integer we can try the following:
+
+```JavaScript
+...
+app.get(/^\/products\/(\d+)$/, function(req, res) {
+    var productId = parseInt(req.params[0], 10);
+    ...
+});
```

Comment:
It would be nice if those code snippets produced something the reader could run and see in the browser to see the results.

Reply:
@rohanreddych, I receive the console message but get this error afterwards: `TypeError: path must be absolute or specify root to res.sendFile`